lib/libc/termios/lib_tccheckfifo.c

```diff
@@ -0,0 +1,109 @@
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <sys/ioctl.h>
+
+#include <termios.h>
+#include <errno.h>
+
+#include <tinyara/serial/tioctl.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Variables
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Variables
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: tcrxavailable
+ *
+ * Descripton:
+ *   calls the ioctl ased on cmd,
+ *	TIOCS_AVAIL: check if there is data in rx fifo.
+ *	TIOCS_READY: check if tranmsit fifo is not full.
+ *	TIOCS_EMPTY: check if transmit fifo is empty.
+ *
+ * Input Parameters:
+ *   fd  - The 'fd' argument is an open file descriptor associated with a terminal.
+ *   cmd - ioctl argument.
+ *
+ * Returned Value:
+ *   Upon successful completion, 0 is returned. Otherwise, -1 is returned.
+ *
+ ****************************************************************************/
+
+int tccheckfifo(int fd, int cmd)
+{
+	int ret;
+
+	switch (cmd) {
+	case TIOCS_AVAIL:
+		ret = ioctl(fd, TIOCS_AVAIL, 0);
+		break;
+
+	case TIOCS_READY:
+		ret = ioctl(fd, TIOCS_READY, 0);
+		break;
+
+	case TIOCS_EMPTY:
+		ret = ioctl(fd, TIOCS_EMPTY, 0);
+		break;
+	default:
+		ret = -EINVAL;
```

Comment:
In `ioctl`, if there is no matched cmd, it returns `-ENOTTY` instead of `-EINVAL`.
Which is correct?